src/lib/update-alternatives/model/alternative.rb

```diff
@@ -95,6 +104,26 @@ def empty?
       false
     end
 
+    def choice(new_choice_path)
+      unless choices.map(&:path).include?(new_choice_path)
+        raise "The alternative doesn't have any choice with path '#{new_choice_path}'"
+      end
+      @value = new_choice_path
+      @status = "manual"
+      @modified = true
+    end
+
+    def automatic_mode
```

Comment:
Same problem with the name, nothing indicates that it's modifying the object. A better name would be `set_auto` or simply `automatic_mode!`